Controllers/ProjectControllers/Admin/UserController.php

```diff
@@ -0,0 +1,204 @@
+    {
+        if (!$this->userModel->isSigned()) {
+            if ($this->postInfo->isPost()) {
+                $emptyResult = $this->dataValidator->emptyCheck($this->postInfo->getPostData(), 'registration');
+                if (!in_array(false, $emptyResult)) {
+                    $correctResult = $this->dataValidator->correctCheck($this->postInfo->getPostData());
+                    if (!in_array('', $correctResult)) {
+                        if ($this->fileInfo->isImageSent()) {
+                            if ($this->fileInfo->isImageType()) {
+                                if ($this->fileInfo->isSizeCorrect($params[0])) {
+                                    if (!$this->userModel->newUser($params[0], $correctResult)) {
+                                        $this->adminLocation('/user/registration');
+                                    } else {
+                                        $this->adminHomeLocation();
+                                    }
+                                } else {
+                                    $this->msgModel->setMsg($this->msgModel->notCorrectData('image_size'));
+                                    $this->adminLocation('/user/registration');
+                                }
+                            } else {
+                                $this->msgModel->setMsg($this->msgModel->notCorrectData('image_type'));
+                                $this->adminLocation('/user/registration');
+                            }
+                        } else {
+                            if (!$this->userModel->newUser($params[0], $correctResult)) {
+                                $this->adminLocation('/user/registration');
+                            } else {
+                                $this->adminHomeLocation();
+                            }
+                        }
+                    } else {
+                        foreach ($correctResult as $key => $value) {
+                            if (!$value) {
+                                $this->msgModel->setMsg($this->msgModel->notCorrectData($key));
+                            }
+                        }
+                        $this->adminLocation('/user/registration');
+                    }
+                } else {
+                    foreach ($emptyResult as $key => $value) {
+                        if (!$value) {
+                            $this->msgModel->setMsg($this->msgModel->emptyDataMsg($key));
+                        }
+                    }
+                    $this->adminLocation('/user/registration');
+                }
+            } else {
+                $this->adminHomeLocation();
+            }
+        } else {
+            $this->homeLocationByCustomerType();
+        }
+    }
+
+    /**
+     * Logout action
+     *
+     * @return void
+     */
+    public function logoutAction(): void
+    {
+        if ($this->userModel->isSigned()) {
+            if ($this->userModel->isAdmin()) {
```

Comment:
I don't think you need to copy two if statements every single time you have to check is user admin or not. Admin function checking should include signed checking as well. As a result you will get rid of useless 'isSigned' checking every time you need to check is it admin or not.
Please rework 'isAdmin' function